tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java

```diff
@@ -80,6 +81,11 @@ private Map<DataAdapterKey<?>, Field> buildCacheOfMutableFields() {
 
     @SuppressWarnings("rawtypes")
     public static String getEmittedDataName(Class<? extends DataAdapterBizlogic> clazz) {
+        if (clazz.getName().contains(BytecodeGen.ENHANCER_BY_GUICE_MARKER)) {
```

Comment:
Let's also mention why this change was needed in comments

Reply:
Added the comment